File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.py
class Solution(object):
 def getAverages(self, nums, k):
    n = len(nums)
    ans = [-1] * n
    sum_val = sum(nums[i] for i in range(min(2*k+1, n)))

    range_size = 2*k+1
    for i in range(k, n-k):
        ans[i] = sum_val // range_size
        sum_val -= nums[i-k]
        if i+k+1 < n:
            sum_val += nums[i+k+1]
    return ans
        

File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int n=nums.size();
        vector<int>ans(n,-1);
        long long sum=0;
        for(int i=0;i<min(2*k+1,n);i++)
            sum+=nums[i];


        int rangeSize=2*k+1;
        for(int i=k;i<n-k;i++){
            ans[i]=(int)(sum/rangeSize);
            sum-=nums[i-k];
            if(i+k+1<n)
            sum+=nums[i+k+1];
        }
        return ans;
        
    }
};